### PR TITLE
Update builder and hasSLSA inmem backend. Change HasSLSA to artifact only.

### PR DESCRIPTION
**Pull request by jeffmendoza**
HasSLSA now only accepts artifacts for subject and buildfrom. Affected files are updated except for the neo4j backend, which is mostly commented out. Builder and hasSLSA in memory backend are updated with ids, internal structs, and backedges.

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]
Nice!

File: pkg/assembler/backends/testing/hasSLSA.go [hunk 1]
```diff
@@ -17,536 +17,230 @@ package testing
...
+func matchSLSAPreds(haves []*model.SLSAPredicate, wants []*model.SLSAPredicateSpec) bool {
+	for _, want := range wants {
+		w := &model.SLSAPredicate{
+			Key:   want.Key,
+			Value: want.Value,
 		}
-
-		if matchOrSkip {
-			collectedHasSLSA = append(collectedHasSLSA, h)
+		if !slices.Contains(haves, w) {
```
Comment (jeffmendoza) [inline pkg/assembler/backends/testing/hasSLSA.go:96]: Actually, this won't work with the pointers I think.

Comment (jeffmendoza) [reply pkg/assembler/backends/testing/hasSLSA.go]: Updated

File: cmd/graphql_playground/cmd/ingest.go [hunk 2]
```diff
@@ -60,12 +61,11 @@ func ingestData(port int) {
 func ingestScorecards(ctx context.Context, client graphql.Client) {
 	logger := logging.FromContext(ctx)
 
-	tag := "v2.12.0"
 	source := model.SourceInputSpec{
 		Type:      "git",
 		Namespace: "github",
 		Name:      "github.com/tensorflow/tensorflow",
-		Tag:       &tag,
+		Tag:       ptrfrom.String("v2.12.0"),
```
Comment (lumjjb) [inline cmd/graphql_playground/cmd/ingest.go:68]: :+1:

**Review by lumjjb — APPROVED** [review]

File: cmd/graphql_playground/cmd/ingest.go (hunk 2 above)
Comment (pxp928) [reply cmd/graphql_playground/cmd/ingest.go:68]: Oh thats useful!

**Pull request merged**